### PR TITLE
[Bugfix][RoCM] GPT-OSS + Expert Parallel

### PR DESCRIPTION
**Pull request by varun-sundar-rabindranath**
## Purpose
On local source-build using `python setup.py develop` on RoCM, gpt-oss + expert-parallel segfault

Repro command: `vllm serve openai/gpt-oss-120b --data-parallel-size 2 --enable-expert-parallel`

Why: When using expert-parallel, the invalid topk-ids are marked as `-1`. The bitmatrix construction kernel is supposed to ignore this. This is achieved by comparing `topk_id // 32` with `offs` (which are all >= 0) into the bitmatrix at https://github.com/vllm-project/vllm/blob/2a9e3347e9fbefc4bf991b60dc45a8c156d8696f/vllm/model_executor/layers/fused_moe/gpt_oss_triton_kernels_moe.py#L100 . 
The current code has the assumption that `-1 // 32` will evaluate to `-1` and the comparison will fail. However, I see that this may not be the case always. On RoCM, I see `-1 // 32` evaluates to `0` and results in setting the expert `-1 % 32 = 31`st expert.  This is incorrect. 

Fix: The fix is to robustify the `invalid_expert` check by including the topk-ids directly in the masking operation. 

Note: 
1. I haven't tested this on Nvidia GPUs but I don't remember seeing this behaviour. 
2. This PR also transposes the bitmatrix tensor for efficient access by the downstream OpenAI Triton kernels (Based on references in OpenAI triton_kernels)

## Test Plan
`vllm serve openai/gpt-oss-120b --data-parallel-size 2 --enable-expert-parallel` 

eval : `OPENAI_API_KEY=empty python -m gpt_oss.evals --model openai/gpt-oss-120b --eval gpqa --n-threads 512 --reasoning-effort low --base-url http://localhost:8000/v1 `

## Test Result
`vllm serve openai/gpt-oss-120b --data-parallel-size 2 --enable-expert-parallel` starts up fine. i.e. no segfault

eval : `[{'eval_name': 'gpqa', 'model_name': 'openai__gpt-oss-120b-low_temp1.0_20260302_192828', 'metric': 0.6527777777777778}]`

### CODE REVIEW

File: vllm/model_executor/layers/fused_moe/gpt_oss_triton_kernels_moe.py [hunk 1]
```diff
@@ -471,14 +479,19 @@ def make_routing_data(
...
     BLOCK_SIZE_K = 32
 
-    bm_cols = triton.cdiv(num_local_experts, BLOCK_SIZE_K)  # n_bitpacks
+    bm_cols = triton.cdiv(num_local_experts, 32)  # n_bitpacks
     bitmatrix = torch.zeros(
-        (n_rows, bm_cols), dtype=torch.uint32, device=topk_ids.device
+        (bm_cols, triton.cdiv(n_rows, 32) * 32),
+        dtype=torch.uint32,
+        device=topk_ids.device,
     )
```
Comment (varun-sundar-rabindranath) [inline vllm/model_executor/layers/fused_moe/gpt_oss_triton_kernels_moe.py:522]: defensively use `32` directly. aliasing with `BLOCK_SIZE_K` will lead to a wrong definition of bitmatrix if `BLOCK_SIZE_K` changes.

**Review by tlrmchlsmth — COMMENTED** [review]
Nice work tracking this down.

I'd like to understand if this also affects CUDA, or if we could be causing performance regressions in some cases. 

Tracking this down further:
>The current code has the assumption that -1 // 32 will evaluate to -1 and the comparison will fail. However, I see that this may not be the case always. On RoCM, I see -1 // 32 evaluates to 0 and results in setting the expert -1 % 32 = 31st expert. This is incorrect.

This confusion appears to be caused a difference in behavior between triton and python and numpy, so should affect both NVIDIA and AMD GPUs:
https://triton-lang.org/main/python-api/triton-semantics.html#differences-with-numpy

**Review by tlrmchlsmth — APPROVED** [review]
